### DISCUSSION TITLE
Logging outbound requests (dixit all fetch calls)

### DISCUSSION THREAD

**Discussion by SBoudrias** [vercel/next.js#24960] [Help]
Hi everyone,

I'm trying to get visibility around the outbound requests done on the server-side (e.g. `getInitialProps`/`getServerProps`). Those aren't logged by default, and I'm unsure how I could intercept all calls to `fetch()` (which would be `node-fetch` on the server-side.)

Anyone ran into the same issue and figured out a solution?

**Comment by weyert** [comment]
You can instrument your code and use it log the outgoing requests. You could consider using Opentelemetry to add tracing to your application which comes with out of the box `net`, `dns`, `http(s)`-package instrumentation
